Feat/#8

woojin-devv opened the pull request:
### 현재까지 구현된 구조 요약
- 역할 상태 관리 구조: bool → enum Role { mentee, mentor } 전환 완료
   - lib/features/mypage/domain/role_provider.dart
   - 전역 상태로 활용되므로, 위치를 lib/core/shared/state 등으로 이동하는게 좋을 지 논의해봐야할 것 같다.
### 마이페이지 내 역할 토글 (RoleToggleCard) 구현 완료
- 초기 역할 상태: Role.mentor로 설정

**고려해야할 부분** 
- 초기값을 Role.mentor로 변경하여 앱 시작 시 멘토로 설정한 상태 
  - 추후 온보딩에서 Role.mentor, Role.mentee값 받아오는 로직 구현해야함  
- 마이페이지에서 역할 변경 시 피드/매칭 탭에도 상태가 전이됨 → 전역 상태 설계 유지 여부 확인 필요

woojin-devv committed:
refactor: rename getLevelInfo
- 함수명이 역할 명확하게 명시

woojin-devv committed:
refactor: 함수명 변경, 불필요한 import 제거, StatelessWidget → ConsumerWidget 변환

- 함수 명명 규칙 일관성 확보
- 사용되지 않는 import 정리
- 상태 관리 목적에 맞게 위젯을 ConsumerWidget으로 리팩토링

woojin-devv committed:
Merge branch 'feat/#3' of https://github.com/keodam/keodam-fronted-app into feat/#3

woojin-devv committed:
Merge remote-tracking branch 'origin/develop' into feat/#3

woojin-devv committed:
refactor: 마이페이지 메뉴 타이틀 enum으로 통합
- 기존 하드코딩된 타이틀들을 enum으로 대체

woojin-devv committed:
refactor: bool 기반 isMenteeProvider를 Role enum 기반으로 리팩토링

kaelch commented:
domain에 entity, state, provider를 때려 박은 이유가 있을까요?
```
.
├── mentee_level_info.dart
├── mento_level_info.dart
├── mypage_menu_title.dart
├── notification_provider.dart
├── role_provider.dart
├── user_provider.dart
├── user_state.dart
└── user_state.freezed.dart
```

woojin-devv committed:
chore: 주석정리

woojin-devv committed:
Merge branch 'develop' of https://github.com/keodam/keodam-fronted-app into feat/#8

woojin-devv commented:
명확한 이유는 없었는데, data 디렉토리로 따로 빼서 파일 구조 정리해두겠습니다!

woojin-devv committed:
refactor: Mentee 레벨 관련 파일 분리

- mentee_level.dart: MenteeLevelInfo 모델 정의
- mentee_level_table.dart: 온도 구간별 색상 및 이미지 매핑 정의
- get_mentee_level.dart: 현재 온도로부터 MenteeLevelInfo 반환하는 유틸 함수 정의

woojin-devv committed:
refactor : Mento 레벨 관련 파일 분리

- mento_level.dart: MentoLevelInfo 모델 정의
- mento_level_table.dart: 경험치 구간별 색상 및 이미지 매핑 정의
- get_mento_level.dart: 현재 경험치로부터 MentoLevelInfo 반환하는 유틸 함수 정의

woojin-devv committed:
refactor: provider 디렉토리 생성 및 상태 관리 파일 이동

- notification_provider.dart, role_provider.dart, user_provider.dart 파일을 provider/ 디렉토리로 이동

woojin-devv committed:
refactor: user_state.dart 파일 위치 변경

- 기존 domain/에서 data/model/로 이동

woojin-devv committed:
refactor: 파일 구조 변경에 따른 import 경로 수정

- provider 및 model 디렉토리 분리에 따라 관련 파일들의 import 경로 수정

woojin-devv commented:
## 기존 
- domain에 entity, state, provider 한꺼번에 넣어두었던 파일 구조 

## 변경사항
- `provider/` 디렉토리 생성 및 상태 관련 파일 분리
  - `notification_provider.dart`
  - `role_provider.dart`
  - `user_provider.dart`
  
- `user_state.dart` 위치 변경
  - 기존 `domain/`에서 → `data/model/`로 이동
  - 이에 따른 `user_state.freezed.dart`도 함께 이동

- Mentee/Mento 레벨 관련 파일 분리
  - mentee_level.dart / mento_level.dart: 레벨 정보를 담는 모델 클래스 정의
  - mentee_level_table.dart / mento_level_table.dart: 각 레벨에 대응되는 색상, 이미지, 범위 값 정의
  - get_mentee_level.dart / get_mento_level.dart: 현재 degree, exp기반 레벨 반환 

으로 변경해두었습니다.!

## 구조 
```
lib/features/mypage
├── data
│   ├── model
│   │   ├── mentee_level_table.dart
│   │   ├── mentee_level.dart
│   │   ├── mento_level_table.dart
│   │   ├── mento_level.dart
│   │   ├── user_state.dart
│   │   └── user_state.freezed.dart
│   └── util
│       ├── get_mentee_level.dart
│       └── get_mento_level.dart
├── domain
│   ├── model
│   ├── mypage_menu_title.dart
│   └── util
├── presentation
│   ├── screens
│   │   └── mypage_screen.dart
│   └── widgets
│       ├── app_info_tile.dart
│       ├── events_section.dart
│       ├── logout_button.dart
│       ├── logout_section.dart
│       ├── matching_info_tile.dart
│       ├── mentee_degree_card.dart
│       ├── mento_level_card.dart
│       ├── mypage_menu_tile.dart
│       ├── notification_toggle_tile.dart
│       ├── notificiation_section.dart
│       ├── profile_edit_section.dart
│       ├── profile_state_section.dart
│       ├── role_toggle_card.dart
│       ├── section_divider.dart
│       ├── shop_header.dart
│       ├── user_profile_tile.dart
│       └── user_support_section.dart
└── provider
    ├── notification_provider.dart
    ├── role_provider.dart
    └── user_provider.dart

11 directories, 30 files
```
